frontend/packages/kubevirt-plugin/src/components/modals/boot-order-modal/boot-order-modal.tsx

```diff
@@ -0,0 +1,183 @@
+    initialDeviceList.forEach((d) => {
+      // Find the initial device in the updated list.
+      const device = updatedDevices.find(
+        (cd) => d.type === cd.type && d.value.name === cd.value.name,
+      );
+
+      // If a device bootOrder changed, or it was deleted, set alert.
+      if (!device || device.value.bootOrder !== d.value.bootOrder) {
+        setUpdatedAlert(true);
+      }
+    });
+  }, [vmLikeEntity]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  // Re-set device list on isOpen change to true.
+  React.useEffect(() => {
+    if (isOpen) {
+      onReload();
+    }
+  }, [isOpen]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  // Send new bootOrder to k8s.
+  const onSubmit = async (event) => {
+    event.preventDefault();
+
+    // Copy only bootOrder from devices to current device list.
+    const currentDevices = _.cloneDeep(getDevices(vmLikeEntity));
+    devices.forEach((d) => {
+      // Find the device to update.
+      const device = currentDevices.find(
+        (cd) => d.type === cd.type && d.value.name === cd.value.name,
+      );
+
+      // Update device bootOrder.
+      if (device && d.value.bootOrder) {
+        device.value.bootOrder = d.value.bootOrder;
+      }
+      if (device && device.value.bootOrder && !d.value.bootOrder) {
+        delete device.value.bootOrder;
+      }
+    });
+
+    // Filter disks and interfaces from devices list.
+    const disks = [
+      ...currentDevices
+        .filter((source) => source.type === DeviceType.DISK)
+        .map((source) => source.value),
+    ];
+    const interfaces = [
+      ...currentDevices
+        .filter((source) => source.type === DeviceType.NIC)
+        .map((source) => source.value),
+    ];
+
+    // Patch k8s.
+    const patches = [
+      new PatchBuilder('/spec/template/spec/domain/devices/disks')
+        .setOperation(PatchOperation.REPLACE)
+        .setValue(disks)
+        .build(),
+      new PatchBuilder('/spec/template/spec/domain/devices/interfaces')
+        .setOperation(PatchOperation.REPLACE)
+        .setValue(interfaces)
+        .build(),
+    ];
+    const promise = k8sPatch(
+      getVMLikeModel(vmLikeEntity),
+      vmLikeEntity,
+      getVMLikePatches(vmLikeEntity, () => patches),
+    );
+
+    handlePromise(promise)
+      .then(() => setOpen(false))
+      .catch(() => setPatchError(true));
+  };
+
+  const actions = [
+    <Button key="confirm" variant={ButtonVariant.primary} onClick={onSubmit}>
+      Save
+    </Button>,
+    <Button key="cancel" variant={ButtonVariant.link} onClick={() => setOpen(false)}>
+      Cancel
+    </Button>,
+  ];
+
+  return (
+    <Modal
+      title={title}
+      isOpen={isOpen}
+      isSmall
+      onClose={() => setOpen(false)}
+      actions={actions}
+      isFooterLeftAligned
+    >
+      <BootOrder devices={devices} setDevices={setDevices} />
+
+      <div className="kubevirt-boot-order-modal__alerts">
```

Comment:
can we reuse `ModalFooter` from `modals` ?

Reply:
you can enhance it to fit your needs better if you need. Maybe also adding children to it if there will be big feature spread.